vars/buildDockerAndPublishImage.groovy

```diff
@@ -97,42 +101,45 @@ spec:
         steps {
           container('img') {
             script {
-              GIT_COMMIT_REV = sh(returnStdout: true, script: "git log -n 1 --pretty=format:'%h'").trim()
-              GIT_SCM_URL = sh(returnStdout: true, script: "git remote show origin | grep 'Fetch URL' | awk '{print \$3}'").trim()
-              SCM_URI = GIT_SCM_URL.replace("git@github.com:", "https://github.com/")
-              BUILD_DATE = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX").format(new Date())
               sh """
+                  export GIT_COMMIT_REV=\$(git log -n 1 --pretty=format:'%h')
+                  export GIT_SCM_URL=\$(git remote show origin | grep 'Fetch URL' | awk '{print \$3}')
+                  export SCM_URI=\$(echo \$GIT_SCM_URL | awk '{print gensub("git@github.com:","https://github.com/",\$3)}')
+
                   img build \
                       -t ${config.registry}${imageName} \
```

Comment:
Would use `withEnv` these  and then use single quotes for the `sh` argument.

Reply:
how would withEnv make this clearer? i'd still need to run sh, and its only used in this one block

That being said, I can revert now though, since the problem was actually credentials breaking with the wrong type

Reply:
Single-quoting let's you skip the escaping of `$`.

Reply:
OH! expose the one non bash one (which I think i did now) and convert it all to single quotes, yea totally! i thought you were saying the other way around.